addon/-private/fields/attr.ts

```diff
@@ -1,24 +1,62 @@
-import { computed } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
 import { Dict } from '@orbit/utils';
 
-export default function (type: string, options: Dict<unknown> = {}) {
-  options.type = type;
+import Model from '../model';
+
+export default function attr(type: string, options: Dict<unknown> = {}) {
+  function trackedAttr(target: any, key: string, desc: PropertyDescriptor) {
+    let trackedDesc = tracked(target, key, desc);
+    let { get: originalGet, set: originalSet } = trackedDesc;
+
+    let defaultAssigned = new WeakSet();
+
+    function setDefaultValue(record: Model) {
+      let value = record.getAttribute(key);
+      setValue(record, value);
+    }
+
+    function setValue(record: any, value: any) {
```

Comment:
```suggestion
    function setValue(record: Model, value: any) {
```

Won't `record` always be typed as a `Model`?

Reply:
yes